api/pacman-api-admin/src/main/java/com/tmobile/pacman/api/admin/repository/service/AbstractAccountServiceImpl.java

```diff
@@ -297,8 +303,33 @@ public void disableAssetGroup(String pluginName) {
             logger.info("AssetGroup {} status {}", pluginName, updateAssetGroupStatus);
             Integer totalUsers = userPreferencesService.updateDefaultAssetGroup(pluginName);
             logger.info("Total users have been updated with the default asset group. The new count is: {}", totalUsers);
+            removePluginFromAssetGroup("all-sources", pluginName);
         } catch (Exception e) {
             logger.error("Unable to make changes to required asset groups for {}", pluginName);
         }
     }
+
+    private void removePluginFromAssetGroup(String assetGroupName, String pluginName) {
+        try {
+            List<String> indicesToRemove = assetGroupService.fetchIndices(pluginName);
+            if (indicesToRemove.isEmpty()) {
+                logger.info("There are no indices to remove");
+                return;
+            }
+            StringBuilder requestBody = new StringBuilder("{\"actions\":[");
+            for (String index : indicesToRemove) {
+                requestBody.append("{\"remove\":{\"index\":\"").append(index)
+                        .append("\",\"alias\":\"").append(assetGroupName).append("\"}},");
+            }
+            requestBody = new StringBuilder(requestBody.substring(0, requestBody.length() - 1)); // Remove the extra comma
```

Comment:
there is two (at least) different methods to do it better:
1. Instead of using string builder you can use list of strings and then join them using ","
2. StringBuilder has method `deleteCharAt` which in pair with method `length` can be used as
```
sb.deleteCharAt(sb.length() - 1)
```

Reply:
I prefer first one, if you ask me